test/jdk/java/awt/dnd/NonAsciiFilenames/NonAsciiFilenames.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2000, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4187490
+ * @summary Verify that Non-ASCII file names can be dragged and dropped
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual NonAsciiFilenames
+ */
+
+import java.awt.Color;
+import java.awt.datatransfer.DataFlavor;
+import java.awt.dnd.DnDConstants;
+import java.awt.dnd.DropTarget;
+import java.awt.dnd.DropTargetDragEvent;
+import java.awt.dnd.DropTargetDropEvent;
+import java.awt.dnd.DropTargetEvent;
+import java.awt.dnd.DropTargetListener;
+import java.io.File;
+import java.util.AbstractList;
+
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+
+public class NonAsciiFilenames {
+    public static void main(String[] args) throws Exception {
+        String INSTRUCTIONS = """
+                This test must be run on an OS which does not use ISO 8859-1
+                as its default encoding.
+
+                Open a native file browsing application, such as Windows
+                Explorer. Try to find a file whose name uses non-ISO 8859-1
+                characters. You will not be verifying the fix for 4187490 unless
+                you drag a file which contains non-ISO 8859-1 characters. Drag
```

Comment:
Would it be better to provide the tester some examples of non-ISO 8859-1 characters and change the test instructions as below?

```suggestion
                Create a file and name it such that it contains non-ISO 8859-1 characters. (For eg. é, à, ö, €, ¥)
```

Reply:
Good idea. Edited.